From 0f4647b4ac812230842e2fc052cfcdeb1250f67b Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Wed, 9 May 2018 18:29:52 +0300
Subject: [PATCH 0634/1051] dts: marvell: Add support for Marvell ARMADA 8082
 DB-B board

Add support for marvell ARMADA 8kPlus board - 8082 configuration B.
The configuration B switches the default boot device from AP SPI to
AP eMMC with 8-bit data interface. AP SPI is disabled since its MPPs
overlap the eMMC ones.

Change-Id: I8c9ba9ff0524e91558c58732252e0e734dfd971d
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 .../bindings/arm/marvell/armada-8kp.txt       |   2 +
 arch/arm64/boot/dts/marvell/Makefile          |   1 +
 .../boot/dts/marvell/armada-8082-db-B.dts     | 208 ++++++++++++++++++
 3 files changed, 211 insertions(+)
 create mode 100644 arch/arm64/boot/dts/marvell/armada-8082-db-B.dts

diff --git a/Documentation/devicetree/bindings/arm/marvell/armada-8kp.txt b/Documentation/devicetree/bindings/arm/marvell/armada-8kp.txt
index 381731a003bd..a5c51b92a7cf 100644
--- a/Documentation/devicetree/bindings/arm/marvell/armada-8kp.txt
+++ b/Documentation/devicetree/bindings/arm/marvell/armada-8kp.txt
@@ -50,6 +50,8 @@ Boards:
       when the board being used is the Armada-8320 emulator
    - "marvell,armada-8082-db"
       when the board being used is the Armada-8082 DB board
+   - "marvell,armada-8082-db-B"
+      when the board being used is the Armada-8082 DB board confugration "B"
    - "marvell,armada-8084-db"
       when the board being used is the Armada-8084 DB board
    - "marvell,armada-8162-db"
diff --git a/arch/arm64/boot/dts/marvell/Makefile b/arch/arm64/boot/dts/marvell/Makefile
index 68e9df3ba1cc..4f1dbcbd366c 100644
--- a/arch/arm64/boot/dts/marvell/Makefile
+++ b/arch/arm64/boot/dts/marvell/Makefile
@@ -8,6 +8,7 @@ dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-mcbin.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-mcbin-single-shot.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8080-pd.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8082-db.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += armada-8082-db-B.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8084-db.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8160-pd.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8162-db.dtb
diff --git a/arch/arm64/boot/dts/marvell/armada-8082-db-B.dts b/arch/arm64/boot/dts/marvell/armada-8082-db-B.dts
new file mode 100644
index 000000000000..6ff08608e948
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/armada-8082-db-B.dts
@@ -0,0 +1,208 @@
+/*
+ * Copyright (C) 2018 Marvell Technology Group Ltd.
+ *
+ * This file is dual-licensed: you can use it either under the terms
+ * of the GPLv2 or the X11 license, at your option. Note that this dual
+ * licensing only applies to this file, and not this project as a
+ * whole.
+ *
+ *  a) This library is free software; you can redistribute it and/or
+ *     modify it under the terms of the GNU General Public License as
+ *     published by the Free Software Foundation; either version 2 of the
+ *     License, or (at your option) any later version.
+ *
+ *     This library is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ * Or, alternatively,
+ *
+ *  b) Permission is hereby granted, free of charge, to any person
+ *     obtaining a copy of this software and associated documentation
+ *     files (the "Software"), to deal in the Software without
+ *     restriction, including without limitation the rights to use,
+ *     copy, modify, merge, publish, distribute, sublicense, and/or
+ *     sell copies of the Software, and to permit persons to whom the
+ *     Software is furnished to do so, subject to the following
+ *     conditions:
+ *
+ *     The above copyright notice and this permission notice shall be
+ *     included in all copies or substantial portions of the Software.
+ *
+ *     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *     OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/*
+ * Device Tree file for Marvell Armada-8082 Development board platform
+ */
+
+#include "armada-8082-db.dtsi"
+
+/ {
+	model = "Marvell 8082-DB-B board";
+	compatible = "marvell,armada-8082-db-B", "marvell,armada-8082-db",
+		     "marvell,armada-8082",
+		     "marvell,armada-ap810-octa", "marvell,armada-ap810";
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
+	memory@00000000 {
+		device_type = "memory";
+		reg = <0x0 0x0 0x0 0x80000000>;
+	};
+
+	vcc_sd_reg0: regulator@0 {
+		compatible = "regulator-fixed";
+		regulator-type = "voltage";
+		regulator-name = "vcc_1_8";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+		regulator-always-on;
+	};
+
+	aliases {
+		ethernet0 = &cp0_eth0;
+		ethernet1 = &cp0_eth1;
+		ethernet2 = &cp0_eth2;
+		ethernet3 = &cp1_eth0;
+	};
+};
+
+&ap0_sdhci0 {
+	status = "okay";
+	bus-width = <8>;
+	non-removable;
+	mmc-hs400-1_8v;
+	vqmmc-supply = <&vcc_sd_reg0>;
+};
+
+&ap0_xor0 {
+	status = "disabled";
+};
+
+&ap0_xor1 {
+	status = "disabled";
+};
+
+&cp0_sdhci0 {
+	status = "okay";
+	bus-width = <4>;
+	no-1-8-v;
+	non-removable;
+};
+
+&cp0_sata0 {
+	status = "okay";
+	sata-port@1 {
+		reg = <1>;
+		status = "okay";
+		/* Generic PHY, providing serdes lanes */
+		phys = <&cp0_comphy5 1>;
+	};
+};
+
+&cp0_pcie0 {
+	status = "okay";
+	msi-parent =<&ap0_gic_its>;
+
+	/* Generic PHY, providing serdes lanes */
+	phys = <&cp0_comphy0 0
+		&cp0_comphy1 0
+		&cp0_comphy2 0
+		&cp0_comphy3 0>;
+	phy-names = "pcie-phy0", "pcie-phy1", "pcie-phy2", "pcie-phy3";
+};
+
+&cp0_pcie1 {
+	msi-parent =<&ap0_gic_its>;
+};
+
+&cp0_pcie2 {
+	msi-parent =<&ap0_gic_its>;
+};
+
+&cp0_mdio {
+	status = "okay";
+
+	phy0: ethernet-phy@0 {
+		reg = <0>;
+	};
+
+	phy1: ethernet-phy@1 {
+		reg = <1>;
+	};
+};
+
+&cp0_ethernet {
+	status = "okay";
+};
+
+&cp0_eth0 {
+	status = "okay";
+	phy-mode = "10gbase-kr";
+	/* Generic PHY, providing serdes lanes */
+	phys = <&cp0_comphy4 0>;
+};
+
+&cp0_eth1 {
+	status = "okay";
+	phy = <&phy0>;
+	phy-mode = "rgmii-id";
+};
+
+&cp0_eth2 {
+	status = "okay";
+	phy = <&phy1>;
+	phy-mode = "rgmii-id";
+};
+
+&cp1_sata0 {
+	status = "okay";
+	sata-port@1 {
+		reg = <1>;
+		status = "okay";
+		/* Generic PHY, providing serdes lanes */
+		phys = <&cp1_comphy5 1>;
+	};
+};
+
+&cp1_pcie0 {
+	status = "okay";
+	msi-parent =<&ap0_gic_its>;
+
+	/* Generic PHY, providing serdes lanes */
+	phys = <&cp1_comphy0 0
+		&cp1_comphy1 0
+		&cp1_comphy2 0
+		&cp1_comphy3 0>;
+	phy-names = "pcie-phy0", "pcie-phy1", "pcie-phy2", "pcie-phy3";
+};
+
+&cp1_pcie1 {
+	msi-parent =<&ap0_gic_its>;
+};
+
+&cp1_pcie2 {
+	msi-parent =<&ap0_gic_its>;
+};
+
+&cp1_ethernet {
+	status = "okay";
+};
+
+&cp1_eth0 {
+	status = "okay";
+	phy-mode = "10gbase-kr";
+	/* Generic PHY, providing serdes lanes */
+	phys = <&cp1_comphy4 0>;
+};
-- 
2.17.1

